Assets/MRTK/Tests/PlayModeTests/SpherePointerTests.cs

```diff
@@ -109,6 +123,52 @@ public IEnumerator GrabLayerMasks()
             Assert.True(pointer.IsInteractionEnabled);
         }
 
+        /// <summary>
+        /// Verifies that SpherePointer correctly returns IsNearObject and IsInteractionEnabled
+        /// when 
+        /// </summary>
+        [UnityTest]
+        public IEnumerator GrabLayerMasksWithOverlap()
+        {
+            // Initialize hand
+            var rightHand = new TestHand(Handedness.Right);
+            yield return rightHand.Show(Vector3.zero);
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.OpenSteadyGrabPoint);
+
+            var pointer = rightHand.GetPointer<SpherePointer>();
+            Assert.IsNotNull(pointer, "Expected to find SpherePointer in the hand controller");
+            Vector3 interactionEnabledPos = new Vector3(0.05f, 0, colliderSurfaceZ - pointer.SphereCastRadius);
+
+            Assert.False(pointer.IsNearObject);
+            Assert.False(pointer.IsInteractionEnabled);
+
+            //Initialize overlap cube
+            overlapRect.SetActive(true);
+
+            // Set layer to spatial mesh, which sphere pointer should be ignoring
+            // assumption: layer 31 is the spatial mesh layer
+            cube.SetLayerRecursively(31);
```

Comment:
It would be great to verify that without your code fix, that this test fails, by the way. Just to verify that the test catches the fix :-) Can you confirm? This is a tricky case and I want to make sure that with the code broken, the test does fail.

Reply:
Confirmed!